### ISSUE TITLE
Access denied for user after migrating to new host

### ISSUE THREAD

**Issue by ma-lalonde** [frappe/frappe_docker#1501]
## Description of the issue
I am trying to migrate ERPNext on docker from one host to another. I first tried the backup-migrate route without success, so I figured I would straight up copy the volumes and docker compose files 1 to 1. I still have Access Denied errors after logging in via the site URL and trying to access the desk.

So what I did is copy `sites`, `logs`, `db-data`, `redis-queue-data` and `redis-cache-data` in `/var/lib/docker/volumes/` (and also the `erpnext` folder containing my `docker-compose.yml` and `.env` using `tar` and `rsync` to preserve owners and permissions (I think). Both hosts have the same username and Debian 12 OS. The only difference I can think of is that the original host runs behind `Nginx-Proxy-Manager` and the second one runs behind Cloudflare Tunnel. Both compose files have an external `default` network and `Port 8080` is not posted to the host, as it is either accessed by reverse proxy or tunnel.

I have already tried https://github.com/frappe/frappe_docker/blob/main/docs/troubleshoot.md#fixing-mariadb-issues-after-rebuilding-the-container, which runs without errors but still doesn't help. I check that password hashes match in mysql for both servers, tried multiple hosts, including '%', '172.18.0.16' and '172.%.%.%'. No dice.

The error message:
```
Traceback (most recent call last):
  File "apps/frappe/frappe/app.py", line 99, in application
    init_request(request)
  File "apps/frappe/frappe/app.py", line 195, in init_request
    frappe.local.http_request = HTTPRequest()
                                ^^^^^^^^^^^^^
  File "apps/frappe/frappe/auth.py", line 44, in __init__
    self.set_session()
  File "apps/frappe/frappe/auth.py", line 78, in set_session
    frappe.local.login_manager = LoginManager()
                                 ^^^^^^^^^^^^^^
  File "apps/frappe/frappe/auth.py", line 120, in __init__
    self.make_session(resume=True)
  File "apps/frappe/frappe/auth.py", line 210, in make_session
    frappe.local.session_obj = Session(
                               ^^^^^^^^
  File "apps/frappe/frappe/sessions.py", line 220, in __init__
    self.resume()
  File "apps/frappe/frappe/sessions.py", line 296, in resume
    self.validate_user()
  File "apps/frappe/frappe/sessions.py", line 228, in validate_user
    if not frappe.get_cached_value("User", self.user, "enabled"):
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/__init__.py", line 1277, in get_cached_value
    doc = get_cached_doc(doctype, name)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/__init__.py", line 1220, in get_cached_doc
    doc = get_doc(*args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/__init__.py", line 1346, in get_doc
    doc = frappe.model.document.get_doc(*args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/model/document.py", line 85, in get_doc
    return controller(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/model/document.py", line 126, in __init__
    self.load_from_db()
  File "apps/frappe/frappe/model/document.py", line 167, in load_from_db
    d = frappe.db.get_value(
        ^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/database/database.py", line 514, in get_value
    result = self.get_values(
             ^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/database/database.py", line 618, in get_values
    out = self._get_values_from_table(
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/database/database.py", line 891, in _get_values_from_table
    return query.run(as_dict=as_dict, debug=debug, update=update, run=run, pluck=pluck)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/query_builder/utils.py", line 87, in execute_query
    result = frappe.db.sql(query, params, *args, **kwargs)  # nosemgrep
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/database/database.py", line 206, in sql
    self.connect()
  File "apps/frappe/frappe/database/database.py", line 111, in connect
    self._conn: "MariadbConnection" | "PostgresConnection" = self.get_connection()
                                                             ^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/database/mariadb/database.py", line 107, in get_connection
    conn = self._get_connection()
           ^^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/database/mariadb/database.py", line 113, in _get_connection
    return self.create_connection()
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/database/mariadb/database.py", line 116, in create_connection
    return pymysql.connect(**self.get_connection_settings())
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "env/lib/python3.11/site-packages/pymysql/connections.py", line 361, in __init__
    self.connect()
  File "env/lib/python3.11/site-packages/pymysql/connections.py", line 669, in connect
    self._request_authentication()
  File "env/lib/python3.11/site-packages/pymysql/connections.py", line 971, in _request_authentication
    auth_packet = self._process_auth(plugin_name, auth_packet)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "env/lib/python3.11/site-packages/pymysql/connections.py", line 1005, in _process_auth
    return _auth.sha256_password_auth(self, auth_packet)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "env/lib/python3.11/site-packages/pymysql/_auth.py", line 188, in sha256_password_auth
    return _roundtrip(conn, data)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "env/lib/python3.11/site-packages/pymysql/_auth.py", line 121, in _roundtrip
    pkt = conn._read_packet()
          ^^^^^^^^^^^^^^^^^^^
  File "env/lib/python3.11/site-packages/pymysql/connections.py", line 775, in _read_packet
    packet.raise_for_error()
  File "env/lib/python3.11/site-packages/pymysql/protocol.py", line 219, in raise_for_error
    err.raise_mysql_exception(self._data)
  File "env/lib/python3.11/site-packages/pymysql/err.py", line 150, in raise_mysql_exception
    raise errorclass(errno, errval)
pymysql.err.OperationalError: (1045, "Access denied for user '_eadb5e8a1219e0d1'@'172.18.0.13' (using password: YES)")
```

Any help or tips welcome, I have visited most websites I could find via search engine.

**Comment by revant** [comment]
> tried multiple hosts, including '%', '172.18.0.16' and '172.%.%.%'. No dice.

How many users for db_name@ are in mariadb?

Keep only 1 user in your mariadb. db_name@%, remove all other like db_name@ip(s)

**Comment by ma-lalonde** [comment]
I tried these hosts sequentially, not all at once.

Actually the funny thing is that there are two users in the working ERPNext, one for '%' and one for the exact docker IP '168.xx.xx.xx'. But in the new non-working, I tried to remove either of the two (the troubleshooting steps gave an error) and the other to 'user'@'%' following the troubleshooting instructions, but had no success. I did restart container and try `docker compose down && docker compose up -d` separately.

**Comment by ma-lalonde** [comment]
Okay, so I tried the following way of importing, inspired from [this post](https://discuss.frappe.io/t/has-anyone-ever-successfully-moved-a-working-erpnext-system-to-another-server-without-using-a-server-image/56893/9).

I created a folder `erpnext` with the `docker-compose.yml` file (and .env file):
```
services:
  backend:
    image: frappe/erpnext:${FRAPPE_VERSION}
    deploy:
      restart_policy:
        condition: on-failure
    volumes:
      - sites:/home/frappe/frappe-bench/sites
      - logs:/home/frappe/frappe-bench/logs

  configurator:
    image: frappe/erpnext:${FRAPPE_VERSION}
    deploy:
      restart_policy:
        condition: none
    entrypoint:
      - bash
      - -c
    # add redis_socketio for backward compatibility
    command:
      - >
        ls -1 apps > sites/apps.txt;
        bench set-config -g db_host $$DB_HOST;
        bench set-config -gp db_port $$DB_PORT;
        bench set-config -g redis_cache "redis://$$REDIS_CACHE";
        bench set-config -g redis_queue "redis://$$REDIS_QUEUE";
        bench set-config -g redis_socketio "redis://$$REDIS_QUEUE";
        bench set-config -gp socketio_port $$SOCKETIO_PORT;
    environment:
      DB_HOST: db
      DB_PORT: "3306"
      REDIS_CACHE: redis-cache:6379
      REDIS_QUEUE: redis-queue:6379
      SOCKETIO_PORT: "9000"
    volumes:
      - sites:/home/frappe/frappe-bench/sites
      - logs:/home/frappe/frappe-bench/logs

 db:
    image: mariadb:10.6
    healthcheck:
      test: mysqladmin ping -h localhost --password=${MYSQL_ROOT_PASSWORD}
      interval: 1s
      retries: 20
    deploy:
      restart_policy:
        condition: on-failure
    command:
      - --character-set-server=utf8mb4
      - --collation-server=utf8mb4_unicode_ci
      - --skip-character-set-client-handshake
      - --skip-innodb-read-only-compressed # Temporary fix for MariaDB 10.6
    environment:
      MYSQL_ROOT_PASSWORD: ${MYSQL_ROOT_PASSWORD}
    volumes:
      - db-data:/var/lib/mysql

frontend:
    image: frappe/erpnext:${FRAPPE_VERSION}
    depends_on:
      - websocket
    deploy:
      restart_policy:
        condition: on-failure
    command:
      - nginx-entrypoint.sh
    environment:
      BACKEND: backend:8000
      FRAPPE_SITE_NAME_HEADER: ${SITE}
      SOCKETIO: websocket:9000
      UPSTREAM_REAL_IP_ADDRESS: 127.0.0.1
      UPSTREAM_REAL_IP_HEADER: X-Forwarded-For
      UPSTREAM_REAL_IP_RECURSIVE: "off"
      PROXY_READ_TIMEOUT: 120
      CLIENT_MAX_BODY_SIZE: 50m
    volumes:
      - sites:/home/frappe/frappe-bench/sites
      - logs:/home/frappe/frappe-bench/logs

queue-long:
    image: frappe/erpnext:${FRAPPE_VERSION}
    deploy:
      restart_policy:
        condition: on-failure
    command:
      - bench
      - worker
      - --queue
      - long,default,short
    volumes:
      - sites:/home/frappe/frappe-bench/sites
      - logs:/home/frappe/frappe-bench/logs

  queue-short:
    image: frappe/erpnext:${FRAPPE_VERSION}
    deploy:
      restart_policy:
        condition: on-failure
    command:
      - bench
      - worker
      - --queue
      - short,default
    volumes:
      - sites:/home/frappe/frappe-bench/sites
      - logs:/home/frappe/frappe-bench/logs

redis-queue:
    image: redis:6.2-alpine
    deploy:
      restart_policy:
        condition: on-failure
    volumes:
      - redis-queue-data:/data

  redis-cache:
    image: redis:6.2-alpine
    deploy:
      restart_policy:
        condition: on-failure
    volumes:
      - redis-cache-data:/data

  scheduler:
    image: frappe/erpnext:${FRAPPE_VERSION}
    deploy:
      restart_policy:
        condition: on-failure
    command:
      - bench
      - schedule
    volumes:
      - sites:/home/frappe/frappe-bench/sites
      - logs:/home/frappe/frappe-bench/logs

websocket:
    image: frappe/erpnext:${FRAPPE_VERSION}
    deploy:
      restart_policy:
        condition: on-failure
    command:
      - node
      - /home/frappe/frappe-bench/apps/frappe/socketio.js
    volumes:
      - sites:/home/frappe/frappe-bench/sites
      - logs:/home/frappe/frappe-bench/logs

volumes:
  db-data:
  redis-queue-data:
  redis-cache-data:
  sites:
  logs:

networks:
  default:
    name: apps-common
    external: true
```
notice I removed `site-creator` and I have a Cloudflare Tunnel to `http://erpnext-backend-1:8080`. I have other apps correctly running with this tunnel.

Then I went into the mariadb container (`docker exec -it erpnext-db-1 bash`) and ensured permissions and users (based on my backup's `site-config.json`)
```
GRANT ALL PRIVILEGES ON *.* TO 'root'@'%' IDENTIFIED BY 'root_password' WITH GRANT OPTION; FLUSH PRIVILEGES;
CREATE USER 'db_name'@'%'; FLUSH PRIVILEGES;
SET PASSWORD FOR 'db_name'@'%' = PASSWORD('db_password'); FLUSH PRIVILEGES;
GRANT ALL PRIVILEGES ON *.* TO 'db_name'@'%' IDENTIFIED BY 'db_password' WITH GRANT OPTION; FLUSH PRIVILEGES;
```
I probably went overboard with that last line's permissions, but anyway.

Then I copied my backup to the backend container and executed:
```
gunzip date-site-database.sql.gz
tar xvf public_files.tar.gz
tar xvf private_files.tar.gz
bench new-site --source_sql /home/frappe/frappe-bench/site/private/backups/20241030_115548-site-database.sql --no-mariadb-socket --admin-password=admin_password --db-root-password=root_password --install-app erpnext --set-default site
cp backup/public_files sites/site/public/files
cp backup/private_files sites/site/private/files
```
I still had permission denied for db_name, so I went to the mariadb container and ran the following mysql:
```
GRANT ALL PRIVILEGES ON `db_name`.* TO '_eadb5e8a1219e0d1'@'%' IDENTIFIED BY 'db_password' WITH GRANT OPTION; FLUSH PRIVILEGES;
```

Then I could login and check "my account", but as soon as I try to reach the desk,I get this error log:
```
Traceback (most recent call last):
  File "apps/frappe/frappe/app.py", line 99, in application
    init_request(request)
  File "apps/frappe/frappe/app.py", line 195, in init_request
    frappe.local.http_request = HTTPRequest()
                                ^^^^^^^^^^^^^
  File "apps/frappe/frappe/auth.py", line 44, in __init__
    self.set_session()
  File "apps/frappe/frappe/auth.py", line 78, in set_session
    frappe.local.login_manager = LoginManager()
                                 ^^^^^^^^^^^^^^
  File "apps/frappe/frappe/auth.py", line 120, in __init__
    self.make_session(resume=True)
  File "apps/frappe/frappe/auth.py", line 210, in make_session
    frappe.local.session_obj = Session(
                               ^^^^^^^^
  File "apps/frappe/frappe/sessions.py", line 220, in __init__
    self.resume()
  File "apps/frappe/frappe/sessions.py", line 296, in resume
    self.validate_user()
  File "apps/frappe/frappe/sessions.py", line 228, in validate_user
    if not frappe.get_cached_value("User", self.user, "enabled"):
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/__init__.py", line 1277, in get_cached_value
    doc = get_cached_doc(doctype, name)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/__init__.py", line 1220, in get_cached_doc
    doc = get_doc(*args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/__init__.py", line 1346, in get_doc
    doc = frappe.model.document.get_doc(*args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/model/document.py", line 85, in get_doc
    return controller(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/model/document.py", line 126, in __init__
    self.load_from_db()
  File "apps/frappe/frappe/model/document.py", line 167, in load_from_db
    d = frappe.db.get_value(
        ^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/database/database.py", line 514, in get_value
    result = self.get_values(
             ^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/database/database.py", line 618, in get_values
    out = self._get_values_from_table(
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/database/database.py", line 891, in _get_values_from_table
    return query.run(as_dict=as_dict, debug=debug, update=update, run=run, pluck=pluck)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/query_builder/utils.py", line 87, in execute_query
    result = frappe.db.sql(query, params, *args, **kwargs)  # nosemgrep
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/database/database.py", line 206, in sql
    self.connect()
  File "apps/frappe/frappe/database/database.py", line 111, in connect
    self._conn: "MariadbConnection" | "PostgresConnection" = self.get_connection()
                                                             ^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/database/mariadb/database.py", line 107, in get_connection
    conn = self._get_connection()
           ^^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/database/mariadb/database.py", line 113, in _get_connection
    return self.create_connection()
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/database/mariadb/database.py", line 116, in create_connection
    return pymysql.connect(**self.get_connection_settings())
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "env/lib/python3.11/site-packages/pymysql/connections.py", line 361, in __init__
    self.connect()
  File "env/lib/python3.11/site-packages/pymysql/connections.py", line 669, in connect
    self._request_authentication()
  File "env/lib/python3.11/site-packages/pymysql/connections.py", line 979, in _request_authentication
    auth_packet = _auth.caching_sha2_password_auth(self, auth_packet)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "env/lib/python3.11/site-packages/pymysql/_auth.py", line 268, in caching_sha2_password_auth
    pkt = _roundtrip(conn, data)
          ^^^^^^^^^^^^^^^^^^^^^^
  File "env/lib/python3.11/site-packages/pymysql/_auth.py", line 121, in _roundtrip
    pkt = conn._read_packet()
          ^^^^^^^^^^^^^^^^^^^
  File "env/lib/python3.11/site-packages/pymysql/connections.py", line 775, in _read_packet
    packet.raise_for_error()
  File "env/lib/python3.11/site-packages/pymysql/protocol.py", line 219, in raise_for_error
    err.raise_mysql_exception(self._data)
  File "env/lib/python3.11/site-packages/pymysql/err.py", line 150, in raise_mysql_exception
    raise errorclass(errno, errval)
pymysql.err.OperationalError: (1045, "Access denied for user '_eadb5e8a1219e0d1'@'172.18.0.13' (using password: YES)")

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "apps/frappe/frappe/website/serve.py", line 19, in get_response
    endpoint, renderer_instance = path_resolver.resolve()
                                  ^^^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/website/path_resolver.py", line 38, in resolve
    resolve_redirect(self.path, request.query_string)
  File "apps/frappe/frappe/website/path_resolver.py", line 118, in resolve_redirect
    redirects += frappe.get_all(
                 ^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/__init__.py", line 2070, in get_all
    return get_list(doctype, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/__init__.py", line 2045, in get_list
    return frappe.model.db_query.DatabaseQuery(doctype).execute(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/model/db_query.py", line 167, in execute
    if is_virtual_doctype(self.doctype):
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/utils/caching.py", line 121, in site_cache_wrapper
    _SITE_CACHE[func_key][frappe.local.site][func_call_key] = func(*args, **kwargs)
                                                              ^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/model/utils/__init__.py", line 135, in is_virtual_doctype
    return frappe.db.get_value("DocType", doctype, "is_virtual")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/database/database.py", line 514, in get_value
    result = self.get_values(
             ^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/database/database.py", line 618, in get_values
    out = self._get_values_from_table(
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/database/database.py", line 891, in _get_values_from_table
    return query.run(as_dict=as_dict, debug=debug, update=update, run=run, pluck=pluck)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/query_builder/utils.py", line 87, in execute_query
    result = frappe.db.sql(query, params, *args, **kwargs)  # nosemgrep
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/database/database.py", line 206, in sql
    self.connect()
  File "apps/frappe/frappe/database/database.py", line 111, in connect
    self._conn: "MariadbConnection" | "PostgresConnection" = self.get_connection()
                                                             ^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/database/mariadb/database.py", line 107, in get_connection
    conn = self._get_connection()
           ^^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/database/mariadb/database.py", line 113, in _get_connection
    return self.create_connection()
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/database/mariadb/database.py", line 116, in create_connection
    return pymysql.connect(**self.get_connection_settings())
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "env/lib/python3.11/site-packages/pymysql/connections.py", line 361, in __init__
    self.connect()
  File "env/lib/python3.11/site-packages/pymysql/connections.py", line 669, in connect
    self._request_authentication()
  File "env/lib/python3.11/site-packages/pymysql/connections.py", line 979, in _request_authentication
    auth_packet = _auth.caching_sha2_password_auth(self, auth_packet)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "env/lib/python3.11/site-packages/pymysql/_auth.py", line 268, in caching_sha2_password_auth
    pkt = _roundtrip(conn, data)
          ^^^^^^^^^^^^^^^^^^^^^^
  File "env/lib/python3.11/site-packages/pymysql/_auth.py", line 121, in _roundtrip
    pkt = conn._read_packet()
          ^^^^^^^^^^^^^^^^^^^
  File "env/lib/python3.11/site-packages/pymysql/connections.py", line 775, in _read_packet
    packet.raise_for_error()
  File "env/lib/python3.11/site-packages/pymysql/protocol.py", line 219, in raise_for_error
    err.raise_mysql_exception(self._data)
  File "env/lib/python3.11/site-packages/pymysql/err.py", line 150, in raise_mysql_exception
    raise errorclass(errno, errval)
pymysql.err.OperationalError: (1045, "Access denied for user '_eadb5e8a1219e0d1'@'172.18.0.13' (using password: YES)")
```

Any hints for a next step? Or something I did wrong?

**Comment by revant** [comment]
> Please update your documentation

I don't use whatever you've used (docker compose), so if you find anything missing do update the docs. If you don't update someone may find this issue.

(This frappe_docker repo is purely contributed by community. No one is paid to maintain this, it is totally against frappecloud sales, every employee from frappe that ever did anything on this repo, left the company. Please update what is needed, I'll merge.)

Related script: For a case where I needed to move 200 sites from VMs to a AWS EKS cluster I used this script https://github.com/castlecraft/custom_containers/blob/main/docs/migrate-from-vm.md
